components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp

```diff
@@ -239,7 +239,8 @@
                                     <td><fmt:message key="query.id"/><font color="red">*</font></td>
                                     <td>
                                         <select name="queryId" id="queryId"
-                                                onchange="javascript:location.href = 'addOperation.jsp?selectedQueryId='+this.options[this.selectedIndex].value+'&operationName='+document.getElementById('operationName').value+'&operationDesc='+document.getElementById('operationDesc').value+'&enableStreaming='+document.getElementById('enableStreaming').value+'&returnRequestStatus='+document.getElementById('returnRequestStatus').value+'&flag=true&action=<%=action%>';">
+                                                onchange="javascript:location.href =
+                                                        'addOperation.jsp?selectedQueryId='+this.options[this.selectedIndex].value+'&operationName='+document.getElementById('operationName').value+'&operationDesc='+document.getElementById('operationDesc').value+'&enableStreaming='+document.getElementById('enableStreaming').value+'&returnRequestStatus='+document.getElementById('returnRequestStatus').value+'&flag=true&action=<%=Encode.forHtml(action)%>';">
```

Comment:
This should change to forUriComponent(action) method.
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String